components/ratgdo/ratgdo.cpp

```diff
@@ -240,6 +240,9 @@ namespace ratgdo {
             }
         } else if (cmd == Command::MOTION) {
             this->motion_state = MotionState::DETECTED;
+            this->set_timeout("clear_motion", 3000, [=] {
+                this->motion_state = MotionState::CLEAR;
+            });
```

Comment:
Is it possible that it could get detected again within 3s?  If so should the timer be canceled and rescheduled?

Reply:
> canceled and rescheduled?

It is, due to the `"clear_motion"` label.

Reply:
Right, of course https://github.com/esphome/esphome/blob/3e3266fa744dc234883d7f9aa8837e27f74bef3f/esphome/core/scheduler.cpp#L27